src/plugins/visualizations/public/saved_visualizations/_saved_vis.ts

```diff
@@ -159,5 +155,5 @@ export function createSavedVisClass(services: SavedObjectKibanaServices) {
     }
   }
 
-  return SavedVis;
+  return SavedVis as new (opts: Record<string, unknown> | string) => SavedObject;
```

Comment:
do we need casting here?

Reply:
yes, else typescript will complain that we are returning something with a type that is not exported. i used same approach that was used in SavedSearch